lib/parse.mjs

```diff
@@ -37,3 +37,27 @@ export function parseIntText(text) {
 export function parseFloatText(text) {
   return text ? parseFloat(text.replace(/\s+/g, "")) : null;
 }
+
+/**
+ * @param {object} document
+ * @param {string} type
+ * @returns {object|null}
+ */
+export function getLdJsonByType(document, type) {
+  for (const scriptEl of document.querySelectorAll(`script[type="application/ld+json"]`)) {
+    const scriptText = scriptEl.textContent;
+    let scriptData;
+    try { // important, as JSON can be malformed
+      scriptData = JSON.parse(scriptText);
+    } catch (err) {
+      console.error(`Failed to parse JSON, skipping this script and continuing..., scriptText: ${scriptText}`);
+      continue;
+    }
+    if (Array.isArray(scriptData)) { // although unusual, one script can contain multiple objects
```

Comment:
getLdJsonByType important mainly for `although unusual, one script can contain multiple objects` part